airflow-core/src/airflow/ui/src/pages/Connections/ConnectionForm.tsx

```diff
@@ -74,14 +74,21 @@ const ConnectionForm = ({
   const paramsDic = { paramsDict: connectionTypeMeta[selectedConnType]?.extra_fields ?? ({} as ParamsSpec) };
 
   const [formErrors, setFormErrors] = useState(false);
+  const [isExtraFieldsDirty, setIsExtraFieldsDirty] = useState(false);
+  const [initialExtra, setInitialExtra] = useState("");
```

Comment:
Are you sure we need new useStates that are updated via useEffects? 
We already have an `initialConnection` object that we should compare against the `conf: extra`